sympy/assumptions/ask.py

```diff
@@ -1195,7 +1195,8 @@ def _extract_facts(expr, symbol, check_reversed_rel=True):
     if not expr.has(symbol):
         return
     if isinstance(expr, AppliedPredicate):
-        if expr.arg == symbol:
+        if (expr.arg == symbol) or ((expr.func == Q.positive and expr.arg - symbol < 0) or
```

Comment:
I believe this code is correct. Before https://github.com/sympy/sympy/pull/1890 it was probably working only by accident. You can't in general use facts for expressions that contain `symbol` to conclude things about `symbol` itself, so only facts that are just about `symbol` itself should be returned here.